### PR TITLE
Ready Button changes, to not lock in place.

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
image
Overview:
This handles the ready status closer to how ZK handles ready status.
Ready status is now kept track of gadget-side, and only when all players are ready does widget:GameSetup send the ready status to the engine. 

Features:
This means players can reposition their start location after readying up.
Reading up is one-way, players cannot unready.
auto_ready_timer can be set to auto-start a game X seconds after all players join (currently set to 120 seconds).
This timer is displayed in the center of the screen.
A lock in place button appears after readying up.
This lock in place button prevents changes in start location.
This lock in place button can be toggled on or off.

### CODE REVIEW

File: language/en/interface.json [hunk 1]
```diff
@@ -571,6 +571,9 @@
 		"initialSpawn": {
 			"ready": "Ready",
 			"unready": "Unready",
+			"readied": "Readied",
+			"lock": "Press to Lock",
+			"unlock": "Press to Unlock",
```
Comment (WatchTheFort) [inline language/en/interface.json:576]:
```suggestion
			"lock": "Lock Position",
			"unlock": "Unlock Position",
```
To be clear what the lock/unlock is referring to

Comment (KyleAnthonyShepherd) [reply language/en/interface.json]: Makes sense to me

**Comment by KyleAnthonyShepherd** [comment]
image

**Comment by KyleAnthonyShepherd** [comment]
image

**Pull request merged**